test/images/agnhost/netexec/netexec.go

```diff
@@ -161,11 +166,18 @@ func (a *atomicBool) get() bool {
 func main(cmd *cobra.Command, args []string) {
 	exitCh := make(chan shutdownRequest)
 
-	if delayShutdown > 0 {
+	sigTermReceived := make(chan struct{})
+	go func() {
 		termCh := make(chan os.Signal, 1)
 		signal.Notify(termCh, syscall.SIGTERM)
+
+		<-termCh
+		close(sigTermReceived)
+	}()
+
+	if delayShutdown > 0 {
```

Comment:
this is the problem, if not delayShutdown is set the process keeps hanging on sigterm

Reply:
> this is the problem, if not delayShutdown is set the process keeps hanging on sigterm

Which particular line does the process hang on?  I don't mind reverting or fixing, but this line existed gating off this go func before.

Reply:
before, we didn't capture the sigterm signal if delayShutdown was 0.
Now, we always capture the sigterm signal, but if we have delayShutdown == 0, we don't execute the os.Exit() and we don't signal the process to exit